src/Profiles.ts

```diff
@@ -1091,6 +1125,66 @@ export class Profiles {
         return filteredProfile;
     }
 
+    public async getCombinedProfile(serviceProfile: IProfileLoaded, baseProfile: IProfileLoaded) {
+        // TODO: This needs to be improved
+        // The idea is to handle all type of ZE Profiles
+        const commonApi = await ZoweExplorerApiRegister.getInstance().getCommonApi(serviceProfile);
+
+        // This check will handle service profiles that have username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            return serviceProfile;
+        }
+
+        // This check is for optional credentials
+        if (baseProfile && serviceProfile.profile.host && serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host) && baseProfile.profile.port !== serviceProfile.profile.port)) {
+                return serviceProfile;
+        }
+
+        let combinedSession;
+        if (!commonApi.getSessionFromCommandArgument) {
+            // This is here for extenders
+            combinedSession = ZoweExplorerApiRegister.getInstance().getCommonApi(serviceProfile).getSession();
+        } else {
+            // This process combines the information from baseprofile to serviceprofile and create a new session
+            const profSchema = await this.getSchema(serviceProfile.type);
+            const cmdArgs: ICommandArguments = {
+                $0: "zowe",
+                _: [""]
+            };
+            for (const prop of Object.keys(profSchema)) {
+                cmdArgs[prop] = serviceProfile.profile[prop] ? serviceProfile.profile[prop] : baseProfile.profile[prop];
+            }
+            if (baseProfile) {
+                // These fields are specific to base profiles & don't exist on the service profile schema
+                cmdArgs.tokenType = baseProfile.profile.tokenType;
+                cmdArgs.tokenValue = baseProfile.profile.tokenValue;
+            }
+            combinedSession = await zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
```

Comment:
should this line be a call to the new api method getSessionFromCommandArgument()

Reply:
See @phaumer comment above: "Also the else branch when we test for the existence of commonApi.getSessionFromCommandArgument could call the zosmf implementation"

Reply:
@katelynienaber base profiles work with different profile types for all zowe cli plugins and for those plugins the creation of a base profile is available, so I think the api call would be the best way to go because these alternate profiles may have a base profile that will need to work with that information as well and can be handled within their api file.

Reply:
this is in the getSessionFromCommandArgument exists block so any extender that has this method needs their api method to be called to be handled in their own way

Reply:
What I meant was that this line could be something like 
```ts
if (commonApi.getSessionFromCommandArgument) {
  combinedSession = await commonApi.getSessionFromCommandArgument(cmdArgs)
} else {
  combinedSession = await zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
}
```

but now I realize that it is probably not such a good idea as there could be no zosmf available on the other end for the user and it would be better to throw a proper error telling the user that this version of the extension does not support Base Profiles, yet.

Reply:
I think this line 1163 should be `ZoweExplorerApiRegister.getInstance().getCommonApi(serviceProfile).getSessionFromCommandArgument(cmdArgs);` to make use of the new api method that was created

Reply:
this has been changed